definitions/checks/foreman_proxy/check_tftp_storage.rb

```diff
@@ -0,0 +1,50 @@
+module Checks::ForemanProxy
+  class CheckTftpStorage < ForemanMaintain::Check
+    metadata do
+      label :check_tftp_storage
+      description 'Clean old Kernel and initramfs files from tftp-boot'
+      tags :default
+      confine do
+        feature(:foreman_proxy)
```

Comment:
Yes, you are right.  
We cannot go with the 1st option because not possible to query pgsql db from the capsule. 2nd option is bit complex because there would be the need of creating separate job template and also user will need to manually copy foreman-proxy public key to the capsule.
I also tried to make use of reverse proxy `# curl -u admin:redhat https://capsule.example.com:8443/api/settings?search=name=token_duration`  but there is no way we can get satellite admin password or auth certs on the capsule server. 
I think for now we can go with the 3rd option that is limiting the check for satellite only.